src/core/thread/network_data.cpp

```diff
@@ -814,5 +814,39 @@ uint8_t NetworkData::CountBorderRouters(RoleFilter aRoleFilter) const
     return rlocsLength;
 }
 
+bool NetworkData::ContainsBorderRouterWithRloc(uint16_t aRloc16) const
+{
+    bool                contains = false;
+    Iterator            iterator = kIteratorInit;
+    ExternalRouteConfig route;
+    OnMeshPrefixConfig  prefix;
+
+    while (GetNextExternalRoute(iterator, route) == kErrorNone)
```

Comment:
Just to mention that the NAT64 prefix is also counted as BR, which I think is correct.

Reply:
Yes.  Any external route entry (no condition on flags) counts.